src/electrum/server.rs

```diff
@@ -490,6 +501,25 @@ impl Connection {
         Ok(result)
     }
 
+    fn log_rpc_event(&self, entries: &Vec<(&str, Value)>) {
+        let mut log = json!({});
+
+        if let Some(log_map) = log.as_object_mut() {
+            entries.into_iter().for_each(|e| {
+                log_map.insert(e.0.to_string(), e.1.clone());
+            });
+            log_map.insert(
+                "source".to_string(),
+                json!({
+                    "ip": self.addr.ip().to_string(),
+                    "port": self.addr.port(),
+                }),
+            );
+        }
+
+        info!("{}", log);
```

Comment:
Maybe add a prefix to the log message so they're easier to locate? Something like `RPC-LOGGER:{...json...}`

Reply:
Also, I would consider printing this directly to `STDOUT` rather than going through the log macro, so that it doesn't depend on the configured log level and doesn't have the `INFO` prefix (which seems unnecessary, and may contain ANSI coloring sequences)

Reply:
Suggested change applied. Now the log looks like:

````
INFO - NetworkInfo { version: 240100, subversion: "/Satoshi:24.1.0/", relayfee: 1e-5 }
INFO - BlockchainInfo { chain: "regtest", blocks: 500, headers: 500, bestblockhash: "53d198810c3a77721bdec389a6ac85c902b35069330099faea80d82456bed606", pruned: false, verificationprogress: 1.0, initialblockdownload: Some(false) }
INFO - Electrum RPC server running on 127.0.0.1:50001
INFO - REST server running on 127.0.0.1:3000
INFO - [127.0.0.1:57226] connected peer
ELECTRUM-RPC-LOGGER: {"event":"connection established","source":{"ip":"127.0.0.1","port":57226}}
ELECTRUM-RPC-LOGGER: {"event":"rpc request","method":"blockchain.scripthash.get_history","source":{"ip":"127.0.0.1","port":57226}}
ELECTRUM-RPC-LOGGER: {"event":"rpc response","method":"blockchain.scripthash.get_history","payload_size":38,"source":{"ip":"127.0.0.1","port":57226}}
ELECTRUM-RPC-LOGGER: {"event":"connection closed","source":{"ip":"127.0.0.1","port":57226}}
INFO - [127.0.0.1:57226] disconnected peer
```